src/internal/executor/iexecutor.cpp

```diff
@@ -30,32 +30,27 @@ namespace srf::internal::executor {
 IExecutor::IExecutor(std::shared_ptr<Options> options) : m_impl(make_executor(std::move(options))) {}
 IExecutor::IExecutor(std::unique_ptr<system::ISystem> system) : m_impl(make_executor(system::System::unwrap(*system)))
 {
-    CHECK(m_impl);
```

Comment:
This check should probably stay, since it would be an error condition if `make_executor` returned a `nullptr`.

Reply:
A default constructor should probably be used which takes a `unique_ptr<Executor>`, then the check for not nullptr only needs to be on the default constructor.

Reply:
FYI, you can use ` m_impl(CHECK_NOTNULL(make_executor(system::System::unwrap(*system))))` to perform the check inline